### ISSUE TITLE
Add endpoint for joining a project

### ISSUE THREAD

**Issue by joshsmith** [code-corps/deprecated-code-corps-rails-api#13]
This should be an endpoint for contributors.

**Comment by begedin** [comment]
Blocked by #12

**Comment by begedin** [comment]
Based on how #12 is being implemented, we could go in multiple directions with this one. If we want simple RESTful routes, then we'd need

# First approach
- `POST /contributors` - This one creates a new Contributor record, with `project` set to provided project, `user` set to `current_user` and `status` set to "pending". It can be accessible by any logged in user.
- `PATCH /contributors` . This one takes an existing `Contributor` record and sets its status to some other value. At first, it can be used by an admin or owner to approve a pending `Contributor` record. Later, it can also be used to change contribution levels (the value of `status`) for other contributors as well.

This first approach gives more flexibility, but also puts some responsibility on the client. When the user clicks on "apply for a contributor" (or whatever they click on), the client app will need to know this means creating a new contributor record with `project` set to the project currently being viewed.

# Second approach

We have routes such as
- `POST /projects/:id/apply_for_collaboration` (or some other similar route)
- `POST /contributors/:id/approve,
- `POST /contributors/:id/promote`
- etc.

The actions are more atomic this way and the routes are more verbose and descriptive, but I'm not really sure the client will be able to be dumber this way, especially an ember client.

Due to that, I'll go with the first approach for now, but I can shift to the second if necessary.

**Comment by begedin** [comment]
@JoshSmith  I believe this was handled by #70.  We now have a `POST /contributors` route, which enables joining a project.

However, I believe we should change this route so it's nested under the project - `POST /:member_slug/:projet_slug/contributors` to maintain consistency. I'm creating a new issue to address this and closing this one - #107
